server/server.go

```diff
@@ -1068,6 +1071,77 @@ func writeHealthResponse(w http.ResponseWriter, err error, payload interface{})
 	writer.JSON(w, status, payload, false)
 }
 
+func (s *Server) unversionedGetHealthWithPolicy(w http.ResponseWriter, r *http.Request) {
+	pluginStatus := s.manager.PluginStatus()
+	pluginState := map[string]string{}
+
+	// optimistically assume all plugins are ok
+	allPluginsOk := true
+	// iterate over plugin status to extract state
+	for name, status := range pluginStatus {
+		if status != nil {
+			pluginState[name] = string(status.State)
+			// if all plugins have not been in OK state yet, then check to see if plugin state is OKx
+			if !s.allPluginsOkOnce && status.State != plugins.StateOK {
+				allPluginsOk = false
+			}
+		}
+	}
+	// once all plugins are OK, set the allPluginsOkOnce flag to true, indicating that all
+	// plugins have achieved a "ready" state at least once on the server.
+	if allPluginsOk {
+		s.allPluginsOkOnce = true
+	}
+
+	input := map[string]interface{}{
+		"plugin_state":  pluginState,
+		"plugins_ready": s.allPluginsOkOnce,
+	}
```

Comment:
nit: the handler can be invoked concurrently so we'll want to wrap this in a mutex to avoid data races on the `s.allPluginsOkOnce` variable.